lib/apply-extends.js

```diff
@@ -16,7 +16,22 @@ function getPathToDefaultConfig (cwd, pathToExtend) {
   return path.resolve(cwd, pathToExtend)
 }
 
-function applyExtends (config, cwd) {
+function mergeDeep (target, config1, config2) {
```

Comment:
Could it be a smarter way to do this and do we miss any case? When I tried the deep merging code got quite complicated fast so tried to have as simple version as possible, but I'm not sure if I miss any cases?

Reply:
I could not think of a case you missed, and as it is a new feature, there is no risk you are going to break any existing code if you missed something.

Reply:
However, apart from mimicking `Object.assign`, what is the use of having a `target` parameter which is always `{}`? I would drop it.

Reply:
Thanks you are right, I've dropped it.